feat: api v2 event types ordering - user, team, org

dhairyashiil opened the pull request:
## What does this PR do?

<!-- Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change. -->

- Fixes #25176 (GitHub issue number)
- Fixes CAL-6755 (Linear issue number - should be visible at the bottom of the GitHub issue description)

### Video:

https://github.com/user-attachments/assets/fde8d870-a6eb-418c-b80b-77c28a89fd16

























<!-- This is an auto-generated description by cubic. -->
---
## Summary by cubic
Adds opt-in ordering to API v2 event type lists. Supports sortCreatedAt=asc|desc for user, team, and org endpoints, aligning with CAL-6755 and fixing #25176.

- **New Features**
  - Introduced sortCreatedAt (asc|desc) to sort by creation date (by ID); default remains no explicit ordering for backward compatibility.
  - Updated API docs and e2e tests to cover asc, desc, and no-parameter cases.

<sup>Written for commit c10e9a6ad3ac71d3c29a7e2f103ffcf1f712391e. Summary will update automatically on new commits.</sup>

<!-- End of auto-generated description by cubic. -->

dhairyashiil committed:
feat: api-v2-event-types-ordering

dhairyashiil committed:
sort team and org event types

dhairyashiil committed:
revert: remove accidental changes to api-auth.strategy.ts

dhairyashiil commented on apps/api/v2/src/ee/event-types/event-types_2024_06_14/event-types.repository.ts:
```diff
@@ -72,6 +72,11 @@ export class EventTypesRepository_2024_06_14 {
       },
+      orderBy: [
+        {
+          id: "desc",
```
we dont have position in data, please check and confirm

supalarry reviewed (changes requested):
1. Can we update tests in `apps/api/v2/src/ee/event-types/event-types_2024_06_14/controllers/event-types.controller.e2e-spec.ts` e.g. `it(`/GET/event-types by username`, async () => {` to confirm the order returned?

2. We should also document the default behaviour that event types are returned oldest to newest in each endpoint affected e.g. `apps/api/v2/src/ee/event-types/event-types_2024_06_14/controllers/event-types.controller.ts` `async getEventTypes(` has the `ApiOperation` where we can mention this.

Nice changes though and thank you for your PR! : )

dhairyashiil committed:
Merge remote-tracking branch 'upstream/main' into feat/api-v2-event-types-ordering

dhairyashiil committed:
docs: add ordering documentation and test for event types endpoints

- Added test assertion to verify event types are returned in descending order by ID (newest first)
- Added API documentation to user event types endpoint describing default ordering behavior
- Added API documentation to team event types endpoint describing default ordering behavior
- Added API documentation to organization event types endpoints describing default ordering behavior

Addresses PR feedback to document and test the ordering behavior introduced in the API v2 event types ordering feature.

supalarry reviewed (changes requested):
Good stuff! I have some thoughts below - let me know what you think

dhairyashiil committed:
feat: add optional sortCreatedAt parameter to event types endpoints

- Add sortCreatedAt query parameter (SortOrderType: "asc" | "desc") to all event types endpoints
- Define SortOrder enum and SortOrderType in pagination.input.ts for reusability
- When not provided, no explicit ordering is applied (backward compatible)
- Update user, team, and organization event types endpoints
- Add comprehensive e2e tests for all sorting scenarios
- Fix circular dependency in platform-types import
- Thread sortCreatedAt through all service layers
- Use spread pattern for conditional orderBy to avoid empty array issues

Addresses PR feedback to make ordering opt-in rather than changing default behavior

dhairyashiil committed:
Merge branch 'main' into feat/api-v2-event-types-ordering

supalarry reviewed (approved):
Great job ! 🚀
